prometheus/promhttp/http_test.go

```diff
@@ -331,3 +333,55 @@ func TestHandlerTimeout(t *testing.T) {
 
 	close(c.Block) // To not leak a goroutine.
 }
+
+func TestHandlerWithCreatedTimestamps(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	handlerWithCT := HandlerFor(reg, HandlerOpts{EnableCreatedTimestamps: true})
```

Comment:
This test screams for [table test](https://thanos.io/tip/contributing/coding-style-guide.md/#table-tests) using t.Run, to clearly separate test cases (with and without CT) (:  Otherwise it's a little bit chaotic